homeassistant/components/environment_canada/config_flow.py

```diff
@@ -28,10 +28,6 @@ async def validate_input(data):
         weather_data = ECWeather(coordinates=(lat, lon), language=lang)
     await weather_data.update()
 
-    if lat is None or lon is None:
-        lat = weather_data.lat
-        lon = weather_data.lon
-
```

Comment:
This code should stay. What was your thinking for deleting it?

Reply:
I assume that this was only used during import in case the user did not specify these parameters. The defaults in the config schema are the coordinates set in home assistant. I believe these would never be `None` since they are set during onboarding, right?

Reply:
The import tests covered these lines but the user flow tests did not so I proposed to remove this for that reason also.

Reply:
I'm surprised a new test is needed. It did have 100% coverage of `config_flow.py`.

Some of the environment canada stuff works with a station ID, other parts with lat/lon. The good thing is when the user specifies just the station ID then base lib returns the lat/lon of that station ID so all parts of the integration has the config it needs.

Reply:
The yaml tests hit those lines so I just rewrote it for the user step that test.